server/app/controllers/api/ProgramApplicationsApiController.java

```
@@ -110,6 +111,9 @@ public CompletionStage<Result> list(
         .getActiveFullProgramDefinitionAsync(programSlug)
         .thenApplyAsync(
             programDefinition -> {
+              if (programDefinition.programType().equals(ProgramType.EXTERNAL)) {
```

Fix and test lgtm!
However, could we have this done before we retrieve the filters and pagination (lines 103-108)? There is no point in making those computations if the program is external

---

Thanks! Hmm I think this is the earliest we can check for external programs since we have to fetch the program definition first. We could potentially build the filter/pagination after this, but that logic is probably lighter weight than loading the program and also includes validation that will fail fast. Seems like it might be more efficient to do the filter/pagination logic first and potentially skip loading the program rather than vice versa, wdyt?

---

Yeah, I was thinking moving the filters and pagination after the program retrieval. However, you are right too. There is also validation there. This lgtm!

---

Sounds good, thank you!